core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -179,31 +179,31 @@ class GroupCoordinator(val brokerId: Int,
 
         if (group.hasStaticMember(groupInstanceId)) {
           val oldMemberId = group.getStaticMemberId(groupInstanceId)
+          info(s"Static member $groupInstanceId with unknown member id rejoins, assigning new member id $newMemberId, while " +
+            s"old member $oldMemberId will be removed. If group is currently stable, no rebalance will be triggered.")
 
-          if (group.is(Stable)) {
-            info(s"Static member $groupInstanceId with unknown member id rejoins, assigning new member id $newMemberId, while " +
-              s"old member $oldMemberId will be removed. No rebalance will be triggered.")
+          val currentLeader = group.leaderOrNull
+          val member = group.replaceGroupInstance(oldMemberId, newMemberId, groupInstanceId)
```

Comment:
The call to `replaceGroupInstance` also updates the leaderId. If the leader is replaced twice, there will be a different leaderId each time. This is probably harmless, just mentioning it since the behavior is slightly inconsistent. The other option would be to lock in the leaderId when the generation is bumped. The advantage of the approach here is that the leaderId set in `GroupMetadata` always corresponds to one of the existing members, which may be less error-prone.

Reply:
Yes, agree it's a trade-off.